profile-widget/diveprofileitem.cpp

```diff
@@ -278,14 +278,20 @@ DivePercentageItem::DivePercentageItem(const DivePlotDataModel &model, const Div
 
 void DivePercentageItem::replot()
 {
-	int sec = 0;
-
 	// Ignore empty values. a heart rate of 0 would be a bad sign.
 	QPolygonF poly;
+	colors.clear();
 	for (int i = 0, modelDataCount = dataModel.rowCount(); i < modelDataCount; i++) {
-		sec = dataModel.index(i, hDataColumn).data().toInt();
+		int sec = dataModel.index(i, hDataColumn).data().toInt();
 		QPointF point(hAxis.posAtValue(sec), vAxis.posAtValue(64 - 4 * tissueIndex));
 		poly.append(point);
+
+		double value = dataModel.index(i, vDataColumn).data().toDouble();
```

Comment:
comment on the commit message: the second paragraph starts `Do all the calculations in paint()` which obviously isn't what you are doing (and would contradict the first paragraph).